src/daemon/Display.cpp

```diff
@@ -116,17 +118,27 @@ namespace SDDM {
         // Create display server
         switch (m_displayServerType) {
         case X11DisplayServerType:
+#ifdef BUILD_WITH_X11
             if (seat()->canTTY()) {
                 m_terminalId = VirtualTerminal::setUpNewVt();
             }
             m_displayServer = new XorgDisplayServer(this);
+#else
+            qCritical() << "Build without X11 support";
+            QCoreApplication::exit(24);
```

Comment:
IMO this should be unreachable by overriding the type instead of exiting.

Reply:
Do I understand you right? You would like to forcefully set `m_displayServerType` to `WaylandDisplayServerType`?
That way I could drop the else-path.

Reply:
Yes. That way users with unsuitable config still get a working sddm.

Reply:
Fixed. Issuing a warning now, modifying `m_displayServerType` to `WaylandDisplayServerType`, and falling through to initialization.